lib/resolver.content.js

```diff
@@ -257,15 +274,23 @@ export default class PodletClientContentResolver {
                 };
             }
 
-            outgoing.emit(
-                'beforeStream',
-                new Response({
-                    headers: outgoing.headers,
-                    js: utils.filterAssets('content', outgoing.manifest.js),
-                    css: utils.filterAssets('content', outgoing.manifest.css),
-                    redirect: outgoing.redirect,
-                }),
-            );
+            const response = new Response({
+                headers: outgoing.headers,
+                js: utils.filterAssets(
+                    'content',
+                    (assetObjects || outgoing.manifest.js).filter(
+                        (assetObject) => assetObject instanceof AssetJs,
+                    ),
+                ),
+                css: utils.filterAssets(
+                    'content',
+                    (assetObjects || outgoing.manifest.css).filter(
+                        (assetObject) => assetObject instanceof AssetCss,
+                    ),
+                ),
+                redirect: outgoing.redirect,
+            });
+            outgoing.emit('beforeStream', response);
```

Comment:
We use the assets from the 103 early hints from the podlet here when available and fallback to the old manifest based assets in order to remain backwards compatible